pypeit/core/extract.py

```diff
@@ -1517,8 +1519,12 @@ def objfind(image, thismask, slit_left, slit_righ, inmask=None, fwhm=3.0, use_us
         plt.xlabel('Approximate Spatial Position (pixels)')
         plt.ylabel('F/sigma (significance)')
         plt.title(qa_title + ': Slit# {:d}'.format(specobj_dict['SLITID']))
-        viewer, ch = display.show_image(image*(thismask*inmask))
-        plt.show()
+        if debug_all:
+            viewer, ch = display.show_image(image*(thismask*inmask))
```

Comment:
I'm also not following the logic here. Is the issue whether or not the plotting is interactive? Note that you can save the file to a figure, and also show it. In other words, you could simply have

plt.title(qa_title + ': Slit# {:d}'.format(specobj_dict['SLITID']))
if outfile_objprof is not None: 
    plt.savefig(outfile_objprof, dpi=400)
if debug_all: 
   viewer, ch = display.show_image(image*(thismask*inmask))
   plt.show()

plt.close('all)